clients/js/src/AdminClient.ts

```diff
@@ -0,0 +1,217 @@
+    }
+
+    /**
+     * Creates a new tenant with the specified properties.
+     *
+     * @param {Object} params - The parameters for creating a new tenant.
+     * @param {string} params.name - The name of the tenant.
+     *
+     * @returns {Promise<Tenant>} A promise that resolves to the created tenant.
+     * @throws {Error} If there is an issue creating the tenant.
+     *
+     * @example
+     * ```typescript
+     * await adminClient.createTenant({
+     *   name: "my_tenant",
+     * });
+     * ```
+     */
+    public async createTenant({
+        name,
+    }: {
+        name: string,
+    }): Promise<Tenant> {
+        const newTenant = await this.api
+            .createTenant({name}, this.api.options)
+            .then(handleSuccess)
+            .catch(handleError);
+
+        // TODO: newTenant is null...... is this a bug in the backend API?
+        // if (newTenant.error) {
+        //     throw new Error(newTenant.error);
+        // }
+
+        return {name: name} as Tenant
+    }
+
+    /**
+     * Gets a tenant with the specified properties.
+     *
+     * @param {Object} params - The parameters for getting a tenant.
+     * @param {string} params.name - The name of the tenant.
+     *
+     * @returns {Promise<Tenant>} A promise that resolves to the tenant.
+     * @throws {Error} If there is an issue getting the tenant.
+     *
+     * @example
+     * ```typescript
+     * await adminClient.getTenant({
+     *   name: "my_tenant",
+     * });
+     * ```
+     */
+    public async getTenant({
+        name,
+    }: {
+        name: string,
+    }): Promise<Tenant> {
+        const getTenant = await this.api
+            .getTenant(name, this.api.options)
+            .then(handleSuccess)
+            .catch(handleError);
+
+        if (getTenant.error) {
+            throw new Error(getTenant.error);
+        }
+
+        return {name: getTenant.name} as Tenant
+    }
+
+    /**
+     * Creates a new database with the specified properties.
+     *
+     * @param {Object} params - The parameters for creating a new database.
+     * @param {string} params.name - The name of the database.
+     * @param {string} params.tenantName - The name of the tenant.
+     *
+     * @returns {Promise<Database>} A promise that resolves to the created database.
+     * @throws {Error} If there is an issue creating the database.
+     *
+     * @example
+     * ```typescript
+     * await adminClient.createDatabase({
+     *   name: "my_database",
+     *   tenantName: "my_tenant",
+     * });
+     * ```
+     */
+    public async createDatabase({
+        name,
+        tenantName
+    }: {
+        name: string,
+        tenantName: string,
+    }): Promise<Database> {
+        const newDatabase = await this.api
+            .createDatabase(tenantName, {name}, this.api.options)
+            .then(handleSuccess)
+            .catch(handleError);
+
+        // TODO: newDatabase is null...... is this a bug in the backend API?
```

Comment:
Same thing, not supposed to return anything, it could have an error due to the middleware. We don't really have a standardized response model ({"success", "error", "<payload_name>"} etc). We should make one and improve our REST API.